### ISSUE TITLE
Failure to import data from old project into new project

### ISSUE THREAD

**Issue by maximFromThe6ix** [wit-ai/wit#381]
For the description of this issue I am going to copy a description from a stackoverflow post that was not asked by me but the issues is one in the same:

"I have a working Wit.Ai application, and want to export, then import it into another Wit.Ai account. I exported the application into a zip file. And there comes trouble: When trying to import the file back into the new account: "We are creating your instance. Please do not refresh" is displayed for several minutes, then, dissapears, without any success or failure message. And no new project is created...

The project I'm trying to import is not huge at all; the .zip file is 3KB, and, got only 4 files in the entities directory."

original post: http://stackoverflow.com/questions/42279826/wit-ai-the-screen-is-stalled-while-importing-application

**Comment by jfrehner** [comment]
Similar issue here. We tried to upload a zip (with an extended expressions.js-file so as to provide more training data to wit.ai). After uploading the zip, the app was created after some minutes of waiting.
We then downloaded the zip-backup to check if everything was imported and noticed that only a part of the expressions.js-file was imported.
How can this issue be fixed? Unfortunately, not being able to import backup-data seriously limits the usefulness of wit.ai for us…

**Comment by hactar** [comment]
Having the same issue, though with a big app and the files are modified by me before reimporting - so it might me my fault. But the app appears to import without an error message.  If I go look at the import then the entities were imported correctly, but many of the stories are missing. I've emailed the file I'm trying to import to Willy and Laurent. It would be great if the import system was more verbose.

**Comment by swapkh91** [comment]
Facing the same problem, I exported my app, deleted from my account, while creating the same one again I imported the downloaded zip and its stuck at "We are creating your instance. Please do not refresh".

**Comment by AndreasGassmann** [comment]
I experienced the same problem. It seems that the server returns an error after a couple of minutes and only processes the data during that time. I get a 500 (Internal Server Error) with the error message "Something went wrong. We've been notified."

**Comment by JoabMendes** [comment]
+1 This is happening with me event with a untouched .zip.

If it helps, I'm getting this log in my console:
<img width="1227" alt="screen shot 2017-03-29 at 15 30 09" src="https://cloud.githubusercontent.com/assets/829669/24470258/b0adb686-1494-11e7-875e-7be056096323.png">

Remembering that I didn't chance the .zip that was created by the web app.

**Comment by blandinw** [comment]
@AndreasGassmann @JoabMendes @swapkh91 @hactar @jfrehner @maximFromThe6ix are you all trying to import legacy apps (without Stories) ?
I'll make this ticket a feature request to make the import work for older apps.
This is on our roadmap, but with low priority for now.

**Comment by maximFromThe6ix** [comment]
I'm trying to import the data from an app with stories, I wouldn't call it a legacy app

**Comment by hactar** [comment]
@blandinw No, this is a stories based app (though I have manipulated the json files to add more stuff, so it may be my fault, but I have no way of knowing as we don't get any feedback on the import). Check your emails, I've just sent two zips I'm trying to import.

**Comment by JoabMendes** [comment]
@blandinw No, it's a recent one and with stories.

**Comment by blandinw** [comment]
Ok, then it is a bug with import, and bumping to hi-pri.

**Comment by AndreasGassmann** [comment]
@blandinw Same here. I exported the project, modified the json, and then uploaded it again.

**Comment by l5t** [comment]
@AndreasGassmann  just checking did you use this zip command?

`zip FlowBased_SimpleExample.zip FlowBased_SimpleExample/app.json FlowBased_SimpleExample/entities/*.json FlowBased_SimpleExample/actions.json FlowBased_SimpleExample/stories.json FlowBased_SimpleExample/expressions.json`

**Comment by AndreasGassmann** [comment]
I used the command provided in the app.json. The import "worked", it just didn't import everything (it randomly stops after about 5 minutes giving me an internal error (500)).

**Comment by blandinw** [comment]
@AndreasGassmann how many expressions did you have in your `expressions.json`?

**Comment by AndreasGassmann** [comment]
I don't know how many expressions there are. The `expressions.json` file is about 10'000 lines long. I simply exported my app and imported the (untouched) zip file again, getting the following error.
```
Failed to load resource: the server responded with a status of 504 (Server timeout)
common.fa5a69a….js:58 error while importing SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at XMLHttpRequest.i.onreadystatechange
```

**Comment by swapkh91** [comment]
Same with me. I exported the app and then tried to import the same zip file, untouched and it was stuck at "We are creating your instance. Please do not refresh" for a long time and finally failed.

**Comment by blandinw** [comment]
Ha. 504 is different than 500. We should import apps in an async way and send an email when the import job is done.
It would be useful to have a few zip files that cause the bug. Can some of you send an email with subject "GitHub issue #381: cannot import" to $my_first_name at fb.com?

**Comment by AndreasGassmann** [comment]
I know that a 504 and 500 error are not the same. Last week I got a 500, just like described above. Today I get a 504 after exactly 2 minutes. I am sure that it took more than 2 minutes last week, so maybe you changed the timeout? 😄 
I agree that the importing should be done asynchronously. Maybe you could redirect the user to the new app after the upload completed and show a progress bar until the project has been fully imported?

**Comment by blandinw** [comment]
Indeed, we started using new load balancers with a more aggressive timeout this week :-)
We'll implement the async thing, but this is low pri compared to what we're working on.
Can I ask how big your .zip file given that it took 5min to upload? We should probably allow importing values and expressions after an app has been created, to make the initial import more likely to succeed.

**Comment by JoabMendes** [comment]
I could make the upload today, but it still threw an error and stuck on the `/new` page without any message displayed. However, When I loaded the home page the new app was listed there with all the entities that where on my `.zip`.
The log says that the error is something about json parsing, but I had validated all the json files on my `.zip` using http://jsonlint.com/. So, this error is weird.

This is a screenshot from my log:
<img width="1267" alt="screen shot 2017-03-29 at 17 29 29" src="https://cloud.githubusercontent.com/assets/829669/24483189/bb5d5c02-14cc-11e7-95c6-26fbbcf91a45.png">

**Comment by AndreasGassmann** [comment]
@blandinw I sent you 2 zips via email (I hope I used the right email). One of the zips was 15KB, the other one 42KB, so I guess that the upload wasn't an issue, just the processing/importing.

**Comment by sindre** [comment]
+1 same problem with the 2 minute timeout when importing.

**Comment by anthonyhseb** [comment]
Getting HTTP 504 too after a minute or two. My app is not even that big (~10 stories, of ~10 turns each, < 200 expressions). It doesn't matter if I modify the zip file or keep it as downloaded, the request keeps timing out.

**Comment by bpleao** [comment]
Any news on this topic? I'm currently facing the same issue: added expressions to expressions.json and zipped following the right order. When importing, the app is there but most of the expressions are missing.

**Comment by blandinw** [comment]
We'll fix all the reported issues with our import/export feature in the coming months. We want to get back to basics and be very reliable for all features we offer, import/export being one of them. We apologize for the inconvenience and thank you for your patience.

**Comment by imtodor** [comment]
Thanks for the comment @blandinw , but what could you suggest to do in order to be able to use large set of expressions and entity values?
